extensions/answer_summarizers/models.py

```diff
@@ -90,6 +96,52 @@ def _get_top_answers_by_frequency(answers, limit=None):
     ])
 
 
+def _get_top_unresolved_answers_by_frequency(
+        answers_with_classification, limit=None):
+    hashable_answers_with_classification = [{
+        'hashable_answer': _HashableAnswer(ans['answer']),
+        'classification_categorization': ans['classification_categorization']
+    } for ans in answers_with_classification]
+
+    classification_results_dict = {}
+    for ans in hashable_answers_with_classification:
+        frequency = 0
+        if ans['hashable_answer'] in classification_results_dict:
+            frequency = classification_results_dict[ans['hashable_answer']][
+                'frequency']
+        classification_results_dict[ans['hashable_answer']] = {
+            'classification_categorization': (
+                ans['classification_categorization']),
+            'frequency': frequency + 1
+        }
+
+    answers_with_most_recent_classification = [{
+        'answer': ans.answer,
+        'classification_categorization': val['classification_categorization'],
+        'frequency': val['frequency']
+    } for ans, val in classification_results_dict.iteritems()]
+
+    grouped_hashable_answers = itertools.groupby(
+        answers_with_most_recent_classification,
+        operator.itemgetter('classification_categorization'))
+
+    unresolved_answers_with_frequency_list = []
+    for category, answers in grouped_hashable_answers:
+        if category in UNRESOLVED_ANSWER_CLASSIFICATION_CATEGORIES:
+            unresolved_answers_with_frequency_list.extend({
+                'answer': ans['answer'],
+                'frequency': ans['frequency']
+            } for ans in answers)
+
+    unresolved_answers_with_frequency_list.sort(
```

Comment:
So I'm wondering if this is getting a bit more complicated than necessary. What do you think about the following approach:
- Do a pass through all answers, filtering out the irrelevant ones (wrong classification category)
- Stick all of them into a collections.Counter
- Call collections.Counter().most_common(limit)

Reply:
The approach you suggest will work only if each answer has unique classification categorization attached to it throughout the entire answers_list. However, that is not the case here.

Since answers are aggregated across multiple versions of exploration it is possible for answer's classification to change throughout the list. Thus I use following approach:

* Pass through all answers, create a list of unique answer where each answer has its most recent classification categorization. (since we are iterating a list anyway, I thought it might be better to also keep track of their frequency).
* Now group the answers according to their classification categorization and merge the answers which belong to unresolved answers categories.
* Return the limit number of answers from the above list.